.github/workflows/apidiff.yml

```diff
@@ -78,19 +79,48 @@ jobs:
       with:
         name: ${{ steps.baseline.outputs.pkg }}
         path: ${{ matrix.changed }}
-    - name: Compare regenerated code to baseline
+    # Step 1: Run apidiff to detect breaking changes.
+    # Instead of failing the job immediately, we capture the output and set a
+    # flag (`breaking_change_found`) to be used in subsequent steps.
+    - name: Detect breaking changes
+      id: detect
       # Only ignore Go interface additions when the PR is from Librarian (e.g., librarian-20251016T095208Z)
       # as it is likely a new method added to the gRPC client stub interface, which is non-breaking.
       env:
         PR_HEAD_REF: ${{ github.event.pull_request.head.ref }}
       run: |
-        cd ${{ matrix.changed }} && apidiff -m -incompatible ${{ steps.baseline.outputs.pkg }} . > diff.txt
+        cd ${{ matrix.changed }}
+        apidiff -m -incompatible ${{ steps.baseline.outputs.pkg }} . > diff.txt
         if [[ "$PR_HEAD_REF" == librarian-* ]]; then
           sed -i '/: added/d' ./diff.txt
         fi
-        cat diff.txt && ! [ -s diff.txt ]
+        cat diff.txt
+        if [ -s diff.txt ]; then
+          echo "breaking_change_found=true" >> $GITHUB_OUTPUT
+        else
+          echo "breaking_change_found=false" >> $GITHUB_OUTPUT
+        fi
+    # Step 2: Conditionally fail the job for GA APIs.
+    # This step runs only if a breaking change was found in the previous step.
+    # It checks if the changed module is a GA version (e.g., "apiv1", "apiv2").
+    # If it is, the `exit 1` command fails the job, blocking the PR.
+    # For non-GA versions (e.g., "apiv1beta1"), this step is skipped, and the
+    # job continues.
+    - name: Conditionally fail for GA APIs
+      id: ga_check # Add an ID to this step to check its outcome later.
+      if: steps.detect.outputs.breaking_change_found == 'true'
+      run: |
+        if [[ "${{ matrix.changed }}" =~ apiv[0-9]+$ ]]; then
```

Comment:
So this is about whether it's a GA API version (e.g. v1), rather than about whether the library is GA. These days APIs often publish beta API specifications under v1 before toggling it to GA in the service config.

Basically this might be a little more confident in something being GA than is really correct, but it's *at least* better than what we have today, so will reduce toil. In the future we could *potentially* parse the repo-metadata.json file to see if the *library* is GA.